rogue/src/components/magic.rs

```diff
@@ -55,46 +58,62 @@ pub struct Castable {
 }
 
 // Tags a spell component as in the spellbook of some other entity. I.e., the
-// referenced entity can cast the spell.
+// referenced entity can cast the spell, and the spell charges regenerate at a
+// base rate of one tick per turn.
 #[derive(Component, ConvertSaveload, Clone)]
 pub struct InSpellBook {
     pub owner: Entity,
     pub slot: BlessingSlot
 }
 
-// Tracks teh number of charges of a spell that are available, and how far we
+// Tracks the number of charges of a spell that are available, and how far we
 // are into recharging a use of that spell.
 #[derive(Component, Serialize, Deserialize, Clone)]
 pub struct SpellCharges {
     // The maximum number of charges of this spell that can be available.
     pub max_charges: i32,
     // The current number of available charges of this spell.
     pub charges: i32,
-    // The number of turns to recharge one use of this spell.
-    pub regen_time: i32,
+    // The number of turns to recharge one use of this spell, assuming the base
+    // recharge rate of one tick per turn.
+    pub regen_ticks: i32,
     // The number of turns we are into recharging another use of this spell. The
     // spell_charge_system is responsible for incrementing this each turn.
-    pub time: i32
+    pub ticks: i32,
+    // Optional element. Used to modify the recharge rate under certain circumstances.
```

Comment:
We're copying the spell element in too many places, and it's going to be a headache to synchronize them. We should rethink our element system a bit.